### PR TITLE
Add propertyDetails to AARead

### LINKED ISSUE #289: Add a properties method to the AARead class
The `AARead` class should have a `properties` method that returns a list of the values from the `dark.aa.PROPERTIES` dict, one value for each AA in the sequence. It should cache the return value in `self` so it doesn't have to be recomputed if `properties` is called more than once.

### PR DESCRIPTION
**Pull request by bamueh**
Fixes #289

### CODE REVIEW

File: dark/reads.py [hunk 1]
```diff
@@ -190,13 +190,23 @@ class AARead(Read):
...
         return (PROPERTIES.get(aa, NONE) for aa in self.sequence)
 
+    def propertyDetails(self):
+        """
+        Translate an amino acid sequence to properties. Each property of the
+        amino acid gets a value scaled from -1 to 1.
+
+        @return: A list of property dictionaries.
+        """
+        return (PROPERTY_DETAILS.get(aa, NONE) for aa in self.sequence)
```
Comment (terrycojones) [inline dark/reads.py]: You don't need the second argument to `get` here, as `None` is the default. I think I may have introduced that, in the `properties` method, so you could delete it there as well, too. Thanks :-)

File: test/test_reads.py [hunk 2]
```diff
@@ -445,6 +445,54 @@ def testPropertiesCorrectTranslation(self):
...
+                    'aliphaticity': -0.157024793388,
+                    'volume': 0.449101796407,
+                    'polar_req': 0.0487804878049,
+                    'hydropathy': -1.0,
+                    'iep': 1.0,
+                    'hydroxyethilation': -0.51486325802,
+                    'aromaticity': -0.0642673521851,
+                    'hydrogenation': -0.401797175866,
+                    'composition': -0.527272727273
+                }, NONE],
```
Comment (terrycojones) [inline test/test_reads.py]: I think the `NONE` and the other 3 things in the list should all have the same indentation. No need to change it if you don't agree.

**Comment by terrycojones** [comment]
Good to merge, with a couple of possible changes as mentioned.

File: dark/reads.py (hunk 1 above)
Comment (bamueh) [inline dark/reads.py]: ```NONE``` in aa.py is assigned ```NONE = 0x1000```. Are you saying I should get rid of that?

Comment (terrycojones) [inline dark/reads.py]: Ah, shit, I thought it said `None`, not `NONE`. No, you'd better leave it in :-)   If you take it out just to see what happens, some tests should fail - if not our tests should be better!

Comment (bamueh) [inline dark/reads.py]: I've made that mistake when writing the code :) yes, the tests are failing.

**Pull request merged**